python/ray/dag/compiled_dag_node.py

```diff
@@ -636,13 +652,21 @@ async def execute_async(
         return AwaitableDAGOutput(fut, self._dag_output_fetcher)
 
     def teardown(self):
-        """Teardown and cancel all worker tasks for this DAG."""
+        """Teardown and cancel all actor tasks for this DAG. After this
+        function returns, the actors should be available to execute new tasks
+        or compile a new DAG."""
         monitor = getattr(self, "_monitor", None)
         if monitor is not None:
-            monitor.teardown()
+            monitor.teardown(wait=True)
 
     def __del__(self):
-        self.teardown()
+        monitor = getattr(self, "_monitor", None)
+        if monitor is not None:
+            # Teardown asynchronously.
```

Comment:
I think it is because del call ordering is not guaranteed when python interpreter exits